demo/grid-drag-and-drop-demos.html

```diff
@@ -0,0 +1,329 @@
+                return i !== draggedItem;
+              });
+
+              const dropOverItem = e.detail.item;
+              const array = dropOverItem.array || targetGridItems;
+
+              if (e.detail.dropLocation === 'on-top') {
+                dropOverItem.children = (dropOverItem.children || []).concat(draggedItem);
+                draggedItem.array = dropOverItem.children;
+              } else {
+                draggedItem.array = array;
+                const dropIndex = array.indexOf(dropOverItem) + (e.detail.dropLocation === 'below' ? 1 : 0);
+                array.splice(dropIndex, 0, draggedItem);
+              }
+              targetGrid.clearCache();
+            });
+
+          });
+        </script>
+      </template>
+    </vaadin-demo-snippet>
+
+    <h3>Custom Drag Data</h3>
+    <p>
+      Operating with the drag event text data enables you to process drag and drop between different application windows.
+      The default payload the drag event is generated from the visible grid columns and items as a line break
+      separated list of tab-separated values.
+    </p>
+    <p>
+      The <code>detail.data</code> property of the <code>grid-drop</code> event
+      is the payload of the drag and drop operation.
+    </p>
+    <p>
+      To customize the drag event data, override the grid's `formatDragData` function. It receives the list of dragged
+      items as arguments and returns the payload as a string representation.
+    </p>
+    <p>
+      <b>Hint:</b> Try dragging the grid's rows to the grid of the same demo but on another browser window.
+    </p>
+    <vaadin-demo-snippet id='grid-drag-and-drop-demos-drag-data'>
+      <template preserve-content>
+        <div style="display: flex">
+          <vaadin-grid drop-mode="between" rows-draggable>
+            <vaadin-grid-selection-column auto-select></vaadin-grid-selection-column>
+            <vaadin-grid-column path="name.first" header="First name"></vaadin-grid-column>
+            <vaadin-grid-column path="name.last" header="Last name"></vaadin-grid-column>
+            <vaadin-grid-column path="email"></vaadin-grid-column>
+          </vaadin-grid>
+        </div>
+
+        <script>
+          window.addDemoReadyListener('#grid-drag-and-drop-demos-drag-data', function(document) {
+            // Assign an array of user objects as the grid's items
+            const grid = document.querySelector('vaadin-grid');
+            grid.items = Vaadin.GridDemo.users.slice(0);
+
+            grid.formatDragData = function(items) {
+              return items.map(function(item) {
+                return item.email;
+              }).join(',');
+            };
+
+            let draggedItems;
+            grid.addEventListener('grid-dragstart', function(e) {
+              draggedItems = e.detail.items;
+            });
+
+            grid.addEventListener('grid-dragend', function(e) {
+              // Remove the items from the source grid
```

Comment:
The data is lost when dropping outside of any grid. Feels like a bad example in this sense, but I don't know if there's any way for the original browser window to know whether the drop succeeded or not.

Reply:
This feature will be implemented afterwards with drop effects API (the drag end event will have the `dropEffect` property). It will help to some extent.